Refreshing tasks - Failed to notify build listener - Found 0 tasks

Extension Name: vscode-gradle
Extension Version: 3.12.0
OS Version: OSX 12.2.1
VSCode version: 1.67.2

**Describe the bug**


**To Reproduce**
mkdir gradle4-demo && cd gradle4-demo
gradle init
code .
Check output of "Gradle for Java"

**Expected behavior**
No thrown error during startup.
After applying plugins in build.gradle the list of tasks should contain the additional tasks

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Output from "Gradle for Java"**

```
[debug] [gradle-server] Starting server
[debug] [gradle-server] Gradle Server cmd: /Users/XXXX/.vscode/extensions/vscjava.vscode-gradle-3.12.0/lib/gradle-server 54934
[info] [gradle-server] Server started, listening on 54934
[info] Gradle client connected to server
[debug] Refreshing tasks
[info] Java Home: /Users/XXXX/.asdf/installs/java/corretto-11.0.15.9.1
[info] JVM Args: --add-opens,java.base/java.util=ALL-UNNAMED,-XX:+HeapDumpOnOutOfMemoryError,-Xmx1024m,-Dfile.encoding=UTF-8,-Duser.country=DE,-Duser.language=de,-Duser.variant
[info] Gradle User Home: /Users/XXXX/.gradle
[info] Gradle Version: 4.10.3
[info] CONFIGURE SUCCESSFUL in 1s

FAILURE: Build failed with an exception.

* What went wrong:
Failed to notify build listener.
> 'java.lang.Class org.gradle.api.reflect.TypeOf.getConcreteClass()'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
[error] Error getting build for /Users/XXXX/entwicklung/otherSources/gradle4-demo: Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.10.3-bin.zip'.
[error] [gradle-server] Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.10.3-bin.zip'.
[info] Java Home: /Users/XXXX/.asdf/installs/java/corretto-11.0.15.9.1
[info] JVM Args: --add-opens,java.base/java.util=ALL-UNNAMED,-XX:+HeapDumpOnOutOfMemoryError,-Xmx1024m,-Dfile.encoding=UTF-8,-Duser.country=DE,-Duser.language=de,-Duser.variant
[info] Gradle User Home: /Users/XXXX/.gradle
[info] Gradle Version: 4.10.3
[info] CONFIGURE SUCCESSFUL in 0s

FAILURE: Build failed with an exception.

* What went wrong:
Failed to notify build listener.
> 'java.lang.Class org.gradle.api.reflect.TypeOf.getConcreteClass()'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

[error] [gradle-server] Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.10.3-bin.zip'.
BUILD FAILED in 0s
[error] Error getting build for /Users/XXXX/entwicklung/otherSources/gradle4-demo: Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.10.3-bin.zip'.
[info] Found 0 tasks

```

**Does the bug still exist if you disable all other extensions?**
Just enabled "Gradle Language Support - v0.2.3" / "Language Support for Java(TM by Red Hat - v.1.6.0" / "German Language Pack v.1.67.2"

**Additional context**
The problem occurs in company network (with proxy settings) and outside (without proxy settings)

--- a/gradle-plugin/src/main/java/com/microsoft/gradle/GradleProjectModelBuilder.java
+++ b/gradle-plugin/src/main/java/com/microsoft/gradle/GradleProjectModelBuilder.java
@@ -42,8 +42,11 @@ import org.gradle.api.tasks.TaskContainer;
 import org.gradle.api.tasks.testing.Test;
 import org.gradle.internal.classpath.ClassPath;
 import org.gradle.tooling.provider.model.ToolingModelBuilder;
+import org.gradle.util.GradleVersion;
 
 public class GradleProjectModelBuilder implements ToolingModelBuilder {
+
+	private static String MINIMAL_SUPPORTED_PLUGIN_CLOSURE_VERSION = "5.0";
 
 	private Set<GradleTask> cachedTasks = new HashSet<>();
 
@@ -161,6 +164,10 @@ public class GradleProjectModelBuilder implements ToolingModelBuilder {
 	}
 
 	private List<GradleClosure> getPluginClosures(Project project) {
+		if (GradleVersion.version(project.getGradle().getGradleVersion())
+				.compareTo(GradleVersion.version(MINIMAL_SUPPORTED_PLUGIN_CLOSURE_VERSION)) < 0) {
+			return Collections.emptyList();
+		}
 		Convention convention = project.getConvention();
 		ExtensionsSchema extensionsSchema = convention.getExtensionsSchema();
 		List<GradleClosure> closures = new ArrayList<>();

